### DISCUSSION TITLE
Major reduction in package's file size?

### DISCUSSION THREAD

**Discussion by em-777** [DrTimothyAldenDavis/SuiteSparse#856] [General]
I'm on Artix Linux and I recently upgraded to the latest version of the SuiteSparse package. I noticed that the file size of the package dropped from 180MB to just 4MB. The same is true with [the package](https://archlinux.org/packages/extra/x86_64/suitesparse/) on regular Arch Linux. Why is that? Forgive me if this seems like a dumb question, but the massive reduction in file size just seems weird to me.

**Comment by mmuetzel** [comment]
It looks like they split GraphBLAS and LAGraph into a separate package:
https://gitlab.archlinux.org/archlinux/packaging/packages/suitesparse/-/commit/b1759417b4bafca0a187fab8c9a639e8c2291b3f

That would reduce the size of the package for the remaining libraries a lot.

**Comment by DrTimothyAldenDavis** [comment]
GraphBLAS is a big compiled library by default.  I do have an option to make it very small (GRAPHBLAS_COMPACT), in which case I compile the kernels as needed on the fly, via my own JIT.  But the linux distros are still using the full sized GraphBLAS I think.  Before adding the JIT, a compact GraphBLAS would be slow; now it's just as fast once the kernels are compiled.
